service/cziti/sdk.go

```diff
@@ -250,4 +261,78 @@ func GetTransferRates(ctx *CZitiCtx) (int64, int64, bool) { //extern void NF_get
 	C.ziti_get_transfer_rates(ctx.nf, &up, &down)
 
 	return int64(up), int64(down), true
+}
+
+var logFile *os.File //the current, active log file
+var logLevel int //set in InitializeCLogger
+
+func InitializeCLogger(level int) {
+	logLevel = level
+	SetLogLevel(logLevel)
+	initializeLogForToday()
+
+	c.AddFunc("@midnight", initiateRollLog)
+	c.Start()
+}
+
+func initiateRollLog() {
+	var async C.uv_async_t
```

Comment:
this does not look correct to me.
should be 
```
async := (*C.uv_async_t)C.malloc(C.uv_async_t_size)
```
then in callback
need to `uv_close` handle and free inside `close_cb`